### DISCUSSION TITLE
discrete variables

### DISCUSSION THREAD

**Discussion by GBenedett** [snl-dakota/dakota#124] [Q&A]
Hi all,

I'm new in using Dakota, I had a templete file to do a surrogate based optimization, I wanted to modify it a bit, in particular I want to use discrete variables instead of continuous ones, but I get the following error when I run Dakota:

> Error: random_sampling requires active variables.

I am attaching my input file, can someone please help me? Thanks a lot
[dakota_case.in.zip](https://github.com/user-attachments/files/15873238/dakota_case.in.zip)

**Comment by jadamstephens** [comment]
The problem is the active variables view. Your study uses "state" variables, which are not active by default when using objective functions. Either change them to design variables or add the keywords `active state` to the variables specification to override the default behavior.

[More information.](https://snl-dakota.github.io/docs/6.20.0/users/usingdakota/inputfile/variables.html#management-of-mixed-variables-by-method)
